iam/src/providers/githubStarredRepoProvider.ts

```diff
@@ -0,0 +1,151 @@
+  }
+
+  const tokenResponse = tokenRequest.data as GithubTokenResponse;
+
+  return tokenResponse.access_token;
+};
+
+const verifyGithub = async (code: string): Promise<GithubFindMyUserResponse> => {
+  // retrieve user's auth bearer token to authenticate client
+  const accessToken = await requestAccessToken(code);
+
+  // Now that we have an access token fetch the user details
+  const userRequest = await axios.get("https://api.github.com/user", {
+    headers: { Authorization: `token ${accessToken}` },
+  });
+
+  if (userRequest.status != 200) {
+    throw `Get user request returned status code ${userRequest.status} instead of the expected 200`;
+  }
+
+  return userRequest.data as GithubFindMyUserResponse;
+};
+
+const verifyUserGithubRepo = async (
+  userData: GithubFindMyUserResponse,
+  code: string
+): Promise<GithubUserRepoResponseData> => {
+  // retrieve user's auth bearer token to authenticate client
+  const accessToken = await requestAccessToken(code);
+
+  // Once access token is received, fetch user repo data
+  const repoRequest: GithubRepoRequestResponse = await axios.get(
+    `https://api.github.com/users/${userData.login}/repos`,
+    {
+      headers: { Authorization: `token ${accessToken}` },
+    }
+  );
+
+  if (repoRequest.status != 200) {
+    throw `Get repo request returned status code ${repoRequest.status} instead of the expected 200`;
+  }
+
+  // Returns an object containing first instance of a user's repo if it has been starred
+  // by a user other than the repo owner, or the last checked repo with no stars
+  const userRepoStarsCheck = repoRequest.data.find((repo: GithubUserRepoResponseData): GithubUserRepoResponseData => {
+    // Check if the GitHub user is the same as the owner of the repo
+    // and if the stargazer count is gt 1
+    if (userData.id === repo.owner.id && repo.stargazers_count > 1) {
+      return repo;
+      // Check if the GitHub user is the same as the owner of the repo
+      // and if the stargazers count equals 1
+    } else if (userData.id === repo.owner.id && repo.stargazers_count === 1) {
+      // Check if the owner of the repo is the same as the only stargazer
+      // if they're different, return true | if they're the same, return false
+      try {
+        // check if the stargazer's user id is equal to the authenticated user's id
+        async (): Promise<GithubUserRepoResponseData> => {
+          const stargazerData: [] = await axios.get(repo.stargazers_url);
+          // Return the first instance where the stargazer type is a "User"
+          const stargazersItem: Record<string, unknown> = stargazerData.find(
+            (stargazerObject: GithubUserRepoResponseData["owner"]) => {
+              stargazerObject.type === userData.type;
+            }
+          );
+          if (stargazersItem.id !== userData.id) {
+            return repo;
+          }
+        };
```

Comment:
This is an anonymous function that never gets called, hence the code in it never gets called.